client.go

```diff
@@ -0,0 +1,37 @@
+package osrm
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+)
+
+// Client makes a real query to OSRM server
+type Client struct {
+	Transport
+
+	serverURL string
+}
+
+// NewClient creates a client with server url and specific getter
+func NewClient(serverURL string, t Transport) Client {
+	return Client{Transport: t, serverURL: serverURL}
+}
+
+// NewClientWithTimeout creates a client with http.Client
+func NewClientWithTimeout(serverURL string, timeout time.Duration) Client {
+	return NewClient(serverURL, NewDefaultTransport(timeout))
+}
+
+// Serve makes GET request to OSRM server and decodes the given JSON
+func (c Client) Serve(ctx context.Context, in *Request, out Response) error {
```

Comment:
Naming looks confusing here - `Serve` is usually associated with handling some requests - not making one.

Reply:
Agree, `DoRequest` is appreciated.

Reply:
also, the function might be unexported

Reply:
updated